src/compiler/checker.ts

```diff
@@ -4333,6 +4338,12 @@ module ts {
             return !!getPropertyOfType(type, "0");
         }
 
+        function isTupleType(type: Type) : boolean {
+            // Check if a Type exactly implements interface TupleType. Typical typechecking code should rely on

```

Comment:
Refering to [this `interface TupleType extends ObjectType`](https://github.com/Microsoft/TypeScript/blob/c371f1e521eb9cf580bba6e502418a60a542da2c/src/compiler/types.ts#L1424-L1427), but yes, will rewrite it like that in a moment